src/embed_tests/TestDomainReload.cs

```diff
@@ -211,9 +211,9 @@ static void RunAssemblyAndUnload(Assembly assembly, string assemblyName)
             {
                 Console.WriteLine($"[Program.Main] The Proxy object is valid ({theProxy}). Unexpected domain unload behavior");
             }
-            catch (Exception)
+            catch (Exception exception)
```

Comment:
You should only catch the `AppDomainUnloadedException` and make it fail it can read the object.

https://github.com/pythonnet/pythonnet/blob/35cbe55b285923bc7253d545d3ae6f530703f4b3/src/embed_tests/TestDomainReload.cs#L163-L171

Reply:
Fixed.